src/DataTable/DataTable.svelte

```diff
@@ -5,7 +5,7 @@
    * @typedef {import('./DataTableTypes.d.ts').PropertyPath<Row>} DataTableKey<Row=DataTableRow>
    * @typedef {any} DataTableValue
    * @typedef {{
-   *    key: DataTableKey<Row>;
+   *    key: DataTableKey<Row> | string;
```

Comment:
What is the reasoning for this?

Though I also question the necessity for the split into two header types, it feels like there could just be one with an optional `value` property.

(To retain auto completions on properties, this should be `| (string & {})` as on the `DataTableCell` key.)

Reply:
Amended per your suggestion in [3407684](https://github.com/carbon-design-system/carbon-components-svelte/pull/1954/commits/34076843a14db994c6169c49c8d2dae64f31ee38)

It was to allow the keys of empty table headers to *not* be included in rows. See https://github.com/carbon-design-system/carbon-components-svelte/pull/1954/commits/cd90958a3cfe639ab6efd0bbe6ba6b9a13bffdbf

```ts
 const headers = [
    { key: "name", value: "Name" },
    { key: "port", value: "Port" },
    { key: "rule", value: "Rule" },
    { key: "overflow", empty: true }, // Keys in empty table headers should not be required as properties in `row`
  ] as const;
```

Reply:
I guess it is a bit more common that columns without headers have some special content and are less likely to be bound to a specific property.